src/Orleans.Core/Serialization/TypeUtilities.cs

```diff
@@ -14,10 +14,14 @@ namespace Orleans.Serialization
 
     internal static class TypeUtilities
     {
-        internal static bool IsOrleansPrimitive(this TypeInfo typeInfo)
+        internal static bool IsOrleansPrimitive(this Type t)
         {
-            var t = typeInfo.AsType();
-            return typeInfo.IsPrimitive || typeInfo.IsEnum || t == typeof(string) || t == typeof(DateTime) || t == typeof(Decimal) || (typeInfo.IsArray && typeInfo.GetElementType().GetTypeInfo().IsOrleansPrimitive());
+            return t.IsPrimitive ||
+                   t.IsEnum ||
+                   t == typeof(string) ||
+                   t == typeof(DateTime) ||
+                   t == typeof(Decimal) ||
+                   (t.IsArray && t.GetElementType().GetTypeInfo().IsOrleansPrimitive());
```

Comment:
You can probably avoid calling GetTypeInfo() since the extension method is now on `Type`